src/Common/src/Interop/Unix/System.Net.Security.Native/Interop.GssApi.cs

```diff
@@ -0,0 +1,128 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static class GssApi
+    {
+        internal static bool EstablishSecurityContext(
```

Comment:
> Is there a reason it's done this way instead?

The only reason we started down this path with Interop.OpenSsl.cs was to allow sharing of the interop code across libraries (eg. System.Net.HttpClient and System.Net.Security) since the 1:1 functions are not very usable by themselves. So if we are sure there is no other consumer for GSSAPI library so we can move all the logic back to NegotiateStreamPal.Unix.cs (or still keep as separate source file but under System.Net.Security\src)